pytorch_lightning/trainer/progress.py

```diff
@@ -141,19 +142,21 @@ def scheduler_steps(self) -> int:
 
 
 @dataclass
-class TrainingProgress(Progress):
-    """
-    Extends ``Progress`` with training specific attributes
+class TrainingLoopProgress(LoopProgress):
 
-    Args:
-        optimization: Tracks optimization progress
-    """
-    optimization: OptimizationProgress = field(default_factory=OptimizationProgress)
+    optimizer_idx: Optional[int] = field(default_factory=int)
+    num_optimizers: Optional[int] = field(default_factory=int)
+    optimizations: Optional[tuple] = field(default_factory=tuple)
 
+    def __post_init__(self):
+        self.optimizations = tuple(deepcopy(OptimizationProgress()) for _ in range(self.num_optimizers))
```

Comment:
why deepcopy?

```suggestion
        self.optimizations = tuple(OptimizationProgress() for _ in range(self.num_optimizers))
```

Reply:
There were a weird bug with DataClass. The OptimizationProgress() were sharing the same reference.

Reply:
Added a comment.

Reply:
Should instead find the issue. Shouldn't need it

```python
from dataclasses import dataclass, field
from typing import List

@dataclass
class Foo:
    x: int = 0


@dataclass
class Thing:
    n: int = field(default=0, repr=False)
    things: List[Foo] = field(init=False)


    def __post_init__(self):
        self.things = [Foo() for _ in range(self.n)]



t =Thing(n=5)
print(t.things)
# [Foo(x=0), Foo(x=0), Foo(x=0), Foo(x=0), Foo(x=0)]

t.things[0].x += 1
print(t.things)
# [Foo(x=1), Foo(x=0), Foo(x=0), Foo(x=0), Foo(x=0)]
```